app/src/main/java/com/keylesspalace/tusky/FileEmojiCompatConfig.java

```diff
@@ -0,0 +1,170 @@
+import android.support.text.emoji.EmojiCompat;
+import android.support.text.emoji.MetadataRepo;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+/**
+ * A simple implementation of EmojiCompat.Config using typeface files.
+ * Based on:
+ * https://android.googlesource.com/platform/frameworks/support/+/master/emoji/bundled/src/main/java/android/support/text/emoji/bundled/BundledEmojiCompatConfig.java
+ * Changes are marked with comments. Formatting and other simple changes are not always marked.
+ */
+public class FileEmojiCompatConfig extends EmojiCompat.Config {
+    // The class name is obviously changed from the original file
+
+    /**
+     * If loading the file failed, a fallback solution will be used.
+     * In order to create minimum friction for the user, replacing all
+     * emojis will be disabled.
+     */
+    private final boolean fallback;
+
+    /**
+     * Create a new configuration for EmojiCompat
+     * @param path The file name/path of the requested font
+     * @param context The context is needed in order to provide the fallback font
+     */
+    public FileEmojiCompatConfig(@NonNull Context context,
+                                 // NEW
+                                 @NonNull String path) {
+        // This one is obviously new
+        this(context, new File(path));
+    }
+
+    /**
+     * Create a new configuration for EmojiCompat
+     * @param fontFile The File containing the requested font
+     * @param context The context is needed in order to provide the fallback font
+     */
+    public FileEmojiCompatConfig(@NonNull Context context, File fontFile) {
+        super(new FileMetadataLoader(context, fontFile));
+        fallback = fontFile == null || !fontFile.exists();
+    }
+
+    /**
+     * If set to true, EmojiCompat will replace every replacable emoji.
+     * Anyhow it will be ignored if the fallback font is used.
+     * @param replaceAll If you want to replace all emojis
+     * @return This EmojiCompat.Config
+     */
+    @Override
+    public EmojiCompat.Config setReplaceAll(boolean replaceAll) {
+        return super.setReplaceAll(replaceAll && !fallback);
+    }
+
+    /**
+     * This is the MetadataLoader. Derived from BundledMetadataLoader but with
+     * the addition of a custom file name.
+     */
+    private static class FileMetadataLoader implements EmojiCompat.MetadataRepoLoader{
+        private final Context mContext;
+        // NEW
+        private final File file;
+
+        private FileMetadataLoader(@NonNull Context context,
+                                    // NEW
+                                    File fontFile) {
+            // We'll need the context for the fallback solution
+            this.mContext = context.getApplicationContext();
+            // NEW
+            this.file = fontFile;
+        }
+
+
+        // Copied from BundledEmojiCompatConfig
+        @Override
+        @RequiresApi(19)
```

Comment:
no need for this, we are on Api 19

Reply:
It's there because I simply copy-pasted BundledEmojiCompatConfig. But I'll remove it of course